server/src/Persistence.hs

```diff
@@ -162,9 +195,18 @@ syncToBackend File pv = do
     syncFile pv
     end <- Clock.getTime Clock.Monotonic
     let time = Clock.toNanoSecs (Clock.diffTimeSpec end start) `div` 1000000
-    if pcLogSync $ pvConfig pv then
-      logMessage pv $ Text.concat ["It took ", Text.pack $ show time, " ms to synchronize Icepeak on disk."]
-    else return ()
+    when (pcLogSync $ pvConfig pv) $ do
+      logMessage pv $ Text.concat ["It took ", Text.pack $ show time, " ms to synchronize Icepeak on disk."] 
+      when (isJust $ pcJournalFile $ pvConfig pv) $ do
+        jStats <- atomically $ do
+          stats <- readTVar (pvJournalStats pv)
+          writeTVar (pvJournalStats pv) emptyJournalStatistics
+          return stats
+        let jCount = journalCount jStats
+            jAvg = if jCount == 0 then 0 else (journalTime jStats) `div` jCount
```

Comment:
There is no logging at all if the count is `0`. I could even remove the division by 0 check here, because of lazy evaluation (but I think it's better to keep a check, you tell me).

Reply:
Oh, right, missed that other when. In that case, why not just move this let binding into the `when` and remove the check here?